core/src/Platforms/Android/AndroidPlatformProxy.cs

```diff
@@ -135,11 +135,21 @@ public string GetProductName()
             return _isMsal ? "MSAL.Xamarin.Android" : "PCL.Android";
         }
 
+        private readonly ILegacyCachePersistence createLegacyCachePersistence = new AndroidLegacyCachePersistence();
```

Comment:
Why on android is the value re-used when it's created new on every other platform?

Reply:
Good catch, fixed.

Reply:
LegacyCachePersistance is not a singleton, a separate instance is created with each TokenCache. Also on some platforms, such as net45, it has state, so reusing the same instance will have consequences.